src/librustc_infer/traits/util.rs

```diff
@@ -112,35 +114,60 @@ pub fn elaborate_predicates<'tcx>(
 ) -> Elaborator<'tcx> {
     let mut visited = PredicateSet::new(tcx);
     predicates.retain(|pred| visited.insert(pred));
-    Elaborator { stack: predicates, visited }
+    let obligations: Vec<_> =
+        predicates.into_iter().map(|predicate| predicate_obligation(predicate, None)).collect();
```

Comment:
Again, this is `Vec` -> `Vec`, which we should avoid ever doing. I wish I spotted this earlier, oof.